app/controllers/add_user.js

```diff
@@ -0,0 +1,281 @@
+    },
+
+    saveUser() {
+      let _this = this;
+      const loadingView = getOwner(_this)
+        .lookup("component:loading")
+        .append();
+
+      let newUser = _this
+        .get("store")
+        .createRecord("user", _this.getRequestParams());
+
+      if (_this.get("selectedRoleIds.length")) {
+        newUser.set("userRoleIds", _this.get("selectedRoleIds"));
+      } else {
+        newUser.set("userRoleIds", []);
+      }
+
+      _this.saveImage().then(imageData => {
+        if (imageData) {
+          let image = _this.get("store").peekRecord("image", imageData.id);
+          newUser.set("image", image);
+        }
+
+        newUser
+          .save()
+          .then(data => {
+            this.clearFormData();
+            this.transitionToRoute("user_details", data.id);
+          })
+          .catch(xhr => {
+            if (xhr.status === 422) {
+              this.get("messageBox").alert(xhr.responseJSON.errors[0].message);
+            } else {
+              throw xhr;
+            }
+          })
+          .finally(() => loadingView.destroy());
+      });
+    },
+
+    cancelForm() {
+      this.get("messageBox").custom(
+        this.locale("users.cancel_user_warning").string,
+        this.locale("yes").string,
+        () => {
+          Ember.run.later(
+            this,
+            function() {
+              this.transitionToRoute("manage_users");
+              this.clearFormData();
+            },
+            0
+          );
+        },
+        this.locale("no").string
+      );
+    },
+
+    uploadReady() {
+      this.set("isReady", true);
+    },
+
+    uploadStart(e, data) {
+      this.set("uploadedFileDate", data);
+      Ember.$(".loading-image-indicator").show();
+      this.set("loadingPercentage", "Image Uploading ");
+    },
+
+    cancelUpload() {
+      if (this.get("uploadedFileDate")) {
+        this.get("uploadedFileDate").abort();
+      }
+    },
+
+    uploadProgress(e, data) {
+      e.target.disabled = true; // disable image-selection
+      var progress = parseInt((data.loaded / data.total) * 100, 10) || 0;
+      this.set("addPhotoLabel", progress + "%");
+      this.set("loadingPercentage", "Image Uploading " + progress + "%");
+    },
+
+    uploadComplete(e) {
+      e.target.disabled = false; // enable image-selection
+      this.set("uploadedFileDate", null);
+      Ember.$(".loading-image-indicator.hide_image_loading").hide();
+      this.set("loadingPercentage", "Image Uploading ");
+    },
+
+    uploadSuccess(e, data) {
+      var identifier =
+        data.result.version +
+        "/" +
+        data.result.public_id +
+        "." +
+        data.result.format;
+      var newUploadedImage = this.get("store").createRecord("image", {
+        cloudinaryId: identifier,
+        favourite: true
+      });
+      this.set("newUploadedImage", newUploadedImage);
+      this.set("userImageKeys", identifier);
+    },
```

Comment:
All the upload methods must certainly been written over and over again, can we maybe centralise them in a mixin ?